openmmtools/multistate/multistatesampler.py

```diff
@@ -1773,11 +1774,15 @@ def _update_timing(self, iteration_time, partial_total_time, run_initial_iterati
     @staticmethod
     def _display_cuda_devices():
         """Query system nvidia-smi to get available GPUs indices and names in debug log."""
-        # Read nvidia-smi query, should return empty strip if no GPU is found.
-        cuda_query_output = os.popen("nvidia-smi --query-gpu=index,gpu_name --format=csv,noheader").read().strip()
-        # Split by line jump and comma
-        cuda_devices_list = [entry.split(',') for entry in cuda_query_output.split('\n')]
-        logger.debug(f"CUDA devices available: {*cuda_devices_list,}")
+        try:
+            cuda_query_output = subprocess.run("nvidia-smi --query-gpu=gpu_uuid,gpu_name,compute_mode  --format=csv", shell=True, capture_output=True, text=True)
+            # Split by line jump and comma
+            cuda_devices_list = [entry for entry in cuda_query_output.stdout.splitlines()]
+            logger.debug(f"CUDA devices available: {*cuda_devices_list,}")
+            if "Exclusive_Process" in cuda_query_output.stdout:
```

Comment:
I think the string here isn't `Exclusive_Process` but `E. Process`, that's why I suggested maybe just catching `not "Default" in cuda_query_output.stdout` ? But whatever you think works best. As of now, we only support the `Default` compute mode (the other mode left is simply disabled).

Reply:
```
$ nvidia-smi --help-query-gpu
...
"compute_mode"
The compute mode flag indicates whether individual or multiple compute applications may run on the GPU.
"Default" means multiple contexts are allowed per device.
"Exclusive_Process" means only one context is allowed per device, usable from multiple threads at a time.
"Prohibited" means no contexts are allowed per device (no compute apps).
...
```

I was going off what the help string seemed to indicated, I think that `not "Default" in cuda_query_output.stdout` is less clear what we are doing, I will explain in the doc string